macros/utils/table_operations/insert_rows.sql

```diff
@@ -0,0 +1,108 @@
+{% macro insert_rows(table_relation, rows, should_commit=false) %}
+    {% if not table_relation %}
+        {{ elementary.edr_log('Recieved table relation is not valid (make sure elementary models were executed successfully first)') }}
+        {{ return(none) }}
+    {% endif %}
+
+    {% set columns = adapter.get_columns_in_relation(table_relation) -%}
+    {% if not columns %}
+        {% set table_name = elementary.relation_to_full_name(table_relation) %}
+        {{ elementary.edr_log('Could not extract columns for table - ' ~ table_name ~ ' (might be a permissions issue)') }}
+        {{ return(none) }}
+    {% endif %}
+
+    {% set insert_rows_method = elementary.get_config_var('insert_rows_method') %}
+    {% if insert_rows_method == 'max_query_size' %}
+      {% set insert_rows_queries = elementary.get_insert_rows_queries(table_relation, columns, rows) %}
+      {% set queries_len = insert_rows_queries | length %}
+      {% for insert_query in insert_rows_queries %}
+        {% do elementary.debug_log("[%d/%d] Running insert query." % (loop.index, queries_len)) %}
+        {% do dbt.run_query(insert_query) %}
+      {% endfor %}
+    {% elif insert_rows_method == 'batch' %}
+      {% set chunk_size = elementary.get_config_var('insert_rows_batch_size') %}
+      {% for rows_chunk in rows | batch(chunk_size) %}
```

Comment:
I think it's a good practice to use the exact same code if you use the flag to "revert" to the previous method

Reply:
The only difference is using Jinja's built-in `batch()` instead of `split_list_to_chunks()`.
It's also better because it doesn't store in-memory all the chunks duplicated but is rather a generator.
Here's the [source code](https://github.com/pallets/jinja/blob/52843b5cbf635b37a82ac0b6c901921a8ee076ff/src/jinja2/filters.py#L1070).